src/client/pages/Checkout/CheckoutPayment/CheckoutPayment.tsx

```diff
@@ -221,9 +221,9 @@ export const CheckoutPayment = ({
     }
   }, [is3DsError, history, trackOfferEvent])
 
-  const addPaymentToCart = useCallback(async () => {
+  const addPaymentToCart = () => {
```

Comment:
not sure if this is what you meant in the previous pr @robinandeer

Reply:
hmm... not sure. Now it seems like we are missing a dependency in the dependency array for `performCheckout`

I think I commented on the fact that the name of the function doesn't represent what it actually does. One way would be to just skip it and call `setIsPaymentConnected(true)` directly inside `performCheckout`, no?